services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/PortHistorySpec.groovy

```diff
@@ -150,13 +155,69 @@ class PortHistorySpec extends HealthCheckSpecification {
         }
     }
 
+    def "Port history is able to show ANTI_FLAP statistic"() {
+        given: "A direct link"
+        def isl = getTopology().islsForActiveSwitches.find { !it.aswitch }
+        assumeTrue("Unable to find ISL for this test", isl as boolean)
+        def timestampBefore = System.currentTimeMillis()
+
+        when: "Execute port DOWN on the src switch"
+        antiflap.portDown(isl.srcSwitch.dpId, isl.srcPort)
+        Wrappers.wait(WAIT_OFFSET / 2) {
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.FAILED
+        }
+
+        then: "Port history is created on the src switch"
+        Wrappers.wait(WAIT_OFFSET) {
+            Long timestampAfterDown = System.currentTimeMillis()
+            with(northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfterDown)) {
+                assert it.size() == 2
+            }
+        }
+
+        when: "Generate antiflap statistic"
+        def count = 0
+        //confd/templates/wfm/topology.properties.tmpl =>  port.antiflap.stats.dumping.interval.seconds = 60
+        def antiflapDumpingInterval = 60
+        Wrappers.timedLoop(antiflapDumpingInterval * 0.9) {
+            northbound.portUp(isl.srcSwitch.dpId, isl.srcPort)
+            sleep(2000)
+            northbound.portDown(isl.srcSwitch.dpId, isl.srcPort)
+            sleep(2000)
+            count += 1
+        }
+
+        then: "Antiflap statistic is available in port history"
+        Wrappers.wait(antiflapDumpingInterval * 0.1 + WAIT_OFFSET) {
```

Comment:
Will I see the `ANTI_FLAP_PERIODIC_STATS` entry if antiflap was deactivated in less than 60 seconds? This would be more interesting test case than flapping for 50s

Reply:
1. This cannot be true since your test generates antiflap events for `60*0.9` which is less than 60 seconds.
2. Even if its true, then this is wrong behavior. We should regardless log antiflap periodic stats even if it lasted 10 seconds. I don't really understand why we shouldn't

Reply:
1. it's true. ANTI_FLAP_PERIODIC_STATS collects stat every 60 seconds
2. I suppose we shouldn't collect  `ANTI_FLAP_PERIODIC_STATS` after antiflap is deactivated
also this situation is possible only on virtual env, because `antiflap.cooldown=10` and  `antiflapDumpingInterval = 60`
it isn't  possible on staging/production envs where `antiflap.cooldown=60 / antiflap.cooldown=300`

Reply:
I will add VIRTUAL tag and relevant comment to needed tests

Reply:
tag and comment were added

Reply:
tests were refactored and new test was added(` def "System shows antiflap statistic in the ANTI_FLAP_DEACTIVATED event when antiflap is deactivated\
 before collecting ANTI_FLAP_PERIODIC_STATS"()`)

now tests should work fine on stage and virtual envs